File: web/src/models/effort-allocation.model.ts
import { Member } from '@pm/models/member.model.ts';
import { ListRequest, ListResponse } from '@pm/models/common.model.ts';

export interface EaWeeklyMember {
  member: Member;
  efforts: EaWeeklyEffort[];
}

export interface EaWeeklyEffort {
  category: string;
  time: number;
  is_product_time: boolean;
}

export interface GetEaWeeklyReportRequest extends ListRequest {}

export interface GetEaWeeklyReportResponse extends ListResponse<EaWeeklyMember> {}

export interface UploadEaWeeklyReportResponse extends ListResponse<EaWeeklyMember> {}

File: api/models/effortallocation.model.go
package models

import "project-management/constants"

type EaWeeklyMember struct {
	Member  *Member           `json:"member"`
	Efforts []*EaWeeklyEffort `json:"efforts"`
}

type EaWeeklyEffort struct {
	Category      constants.LabelCategory `json:"category"`
	Time          float64                 `json:"time"`
	IsProductTime bool                    `json:"is_product_time"`
}

type GetEaWeeklyReportRequest struct {
	ListRequest
}

type GetEaWeeklyReportResponse struct {
	ListResponse
	Items []EaWeeklyMember `json:"items"`
}

type UploadEaWeeklyReportResponse struct {
	ListResponse
	Items []EaWeeklyMember `json:"items"`
}
